examples/aws/storage-efs/README.md

```diff
@@ -0,0 +1,214 @@
+```
+
+3. Create a security group with an inbound rule that allows inbound NFS traffic for your Amazon EFS mount points.
+    a. Create a security group. 
+    ```
+    security_group_id=$(aws ec2 create-security-group \
+        --group-name MyEfsSecurityGroup \
+        --description "My EFS security group" \
+        --vpc-id $vpc_id \
+        --output text)
+    ```
+
+    b. Create an inbound rule that allows inbound NFS traffic from the CIDR for your cluster's VPC.
+    ```
+    aws ec2 authorize-security-group-ingress \
+        --group-id $security_group_id \
+        --protocol tcp \
+        --port 2049 \
+        --cidr $cidr_range
+    ```
+
+4. Create an Amazon EFS file system for your Amazon EKS cluster.
+```
+file_system_id=$(aws efs create-file-system \
+    --region $CLUSTER_REGION \
+    --performance-mode generalPurpose \
+    --query 'FileSystemId' \
+    --output text)
+```
+
+## 5.0 Create Mount Targets for your cluster
+1. [Optional] If you are re-using an existing EFS Volume, you will first have to delete any old mount targets. You can use the following commands for this - 
+```
+aws efs describe-mount-targets --file-system-id $file_system_id
+aws efs delete-mount-target --mount-target-id <each-id>
+```
+
+2. Determine the IP address of your cluster nodes.
+```
+kubectl get nodes
+```
+
+3. Determine the IDs of the subnets in your VPC and which Availability Zone the subnet is in.
+```
+aws ec2 describe-subnets \
+    --filters "Name=vpc-id,Values=$vpc_id" \
+    --query 'Subnets[*].{SubnetId: SubnetId,AvailabilityZone: AvailabilityZone,CidrBlock: CidrBlock}' \
+    --output table
+```
+
+4. Add mount targets for the subnets that your nodes are in. If are more than 1 nodes in the cluster, you'd run the command once for a subnet in each AZ that you had a node in, replacing subnet-EXAMPLEe2ba886490 with the appropriate subnet ID from the previous command
+```
+aws efs create-mount-target \
+    --file-system-id $file_system_id \
+    --security-groups $security_group_id \
+    --subnet-id subnet-EXAMPLEe2ba886490
+```
+
+
+## 6.0 Using the EFS Storage in Kubeflow
+
+### Sample 1 - Static Provisioning
+[Using this sample from official AWS Docs](https://github.com/kubernetes-sigs/aws-efs-csi-driver/tree/master/examples/kubernetes/multiple_pods) we have provided the required spec files in the sample subdirectory but you can create the PVC another way. 
+
+1. Use the `$file_system_id` you recorded before or use the following command to get the efs filesystem id - 
+```
+aws efs describe-file-systems --query "FileSystems[*].FileSystemId" --output text
+```
+
+2. Now edit the sample/pv.yaml and edit the `volumeHandle` to point to your EFS filesystem
+
+3. Now create the required persistentvolume, persistentvolumeclaim and storageclass resources as -
+```
+kubectl apply -f examples/aws/storage-efs/sample/pv.yaml
+kubectl apply -f examples/aws/storage-efs/sample/pvc.yaml
+kubectl apply -f examples/aws/storage-efs/sample/sc.yaml
```

Comment:
shouldn't storage class be first step ?

Reply:
For dynamic provisioning, yes. For Static Provisioning, it does not seem to matter. Order differs on the AWS Docs - https://docs.aws.amazon.com/eks/latest/userguide/efs-csi.html

and the kubeflow docs - https://www.kubeflow.org/docs/distributions/aws/customizing-aws/storage/#static-provisioning

Reply:
ok, lets keep it the way so that it works for both (static and dynamic).
In this case I will lean more on AWS doc than KubeFlow doc.

Reply:
I think this comment is still applicable. please keep the order that works for both.